### PR TITLE
Update workflow.get_runs & pullrequest.add_to_assignees function signature

### PR DESCRIPTION
**Pull request by sd-kialo**
The `workflow.get_runs` function handles both the objects `NamedUser.NamedUser` and `Branch.Branch` and `str` type for its parameters internally but the type signature only accepts the objects. 
`pullrequest.add_to_assignees` handles `NamedUser.NamedUser` and `str` internally but only accepts `NamedUser.NamedUser`.

This change updates the function signatures to match the types handled internally by the function.

### CODE REVIEW

File: github/PullRequest.py [hunk 1]
```diff
@@ -747,7 +747,7 @@ def merge(
         headers, data = self._requester.requestJsonAndCheck("PUT", f"{self.url}/merge", input=post_parameters)
         return github.PullRequestMergeStatus.PullRequestMergeStatus(self._requester, headers, data, completed=True)
 
-    def add_to_assignees(self, *assignees: github.NamedUser.NamedUser) -> None:
+    def add_to_assignees(self, *assignees: github.NamedUser.NamedUser | str) -> None:
         """
         :calls: `POST /repos/{owner}/{repo}/issues/{number}/assignees <https://docs.github.com/en/rest/reference/issues#assignees>`_
```
Comment (trim21) [inline github/PullRequest.py:752]: can you also update this document link? this link is out of date

Comment (trim21) [inline github/PullRequest.py:752]:
```suggestion
        :calls: `POST /repos/{owner}/{repo}/issues/{number}/assignees <https://docs.github.com/en/free-pro-team@latest/rest/issues/assignees?apiVersion=2022-11-28#list-assignees>`_
```

Comment (sd-kialo) [reply github/PullRequest.py]: Commited suggested change.

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Pull request merged**